test/functional/terminal/buffer_spec.lua

```diff
@@ -255,8 +261,21 @@ describe(':terminal buffer', function()
     command('bdelete!')
   end)
 
-  it('handles wqall', function()
+  it("requires bang (!) to close a running job", function()
+    local cwd = funcs.fnamemodify('.', ':p:~'):gsub([[[\/]*$]], '')
+    local ext_pat = iswin() and '%.EXE' or ''
     eq('Vim(wqall):E948: Job still running', exc_exec('wqall'))
+    matches('^Vim%(bdelete%):E89: term://'..pesc(cwd)..'//%d+:'..nvim_dir..'/tty%-test'..ext_pat..' will be killed %(add %! to override%)$', exc_exec('bdelete'))
+    command('call jobstop(&channel)')
+    retry(nil, nil, function() assert(0 >= eval('jobwait([&channel], 1000)[0]')) end)
```

Comment:
because jobwait() is actually waiting, we don't need retry() now.
```suggestion
    assert(0 >= eval('jobwait([&channel], 1000)[0]'))
```